lib/Sema/TypeCheckPattern.cpp

```diff
@@ -1388,19 +1388,24 @@ bool TypeChecker::coercePatternToType(Pattern *&P, TypeResolution resolution,
             goto recur;
           }
 
-          auto diag = diagnose(EEP->getLoc(),
-                               diag::enum_element_pattern_member_not_found,
-                               EEP->getName().str(), type);
-
-          // If we have an optional type let's try to see if the case
-          // exists in its base type, if so we can suggest a fix-it for that.
-          if (auto baseType = type->getOptionalObjectType()) {
-            if (lookupEnumMemberElement(*this, dc, baseType, EEP->getName(),
-                                        EEP->getLoc()))
-              diag.fixItInsertAfter(EEP->getEndLoc(), "?");
+          // If we have an optional type, let's try to see if the case
+          // exists in its base type and if it does then synthesize an
+          // OptionalSomePattern that wraps the case.
+          if (type->getOptionalObjectType()) {
```

Comment:
You could avoid a little bit of unnecessary work by calling `lookThroughAllOptionalTypes` on the object type you found here.

Please mention in the comment that you're relying on recursion in order to add multiple levels of some-pattern if the optional is nested.

Reply:
@rjmccall I thought about doing that, but `lookThroughAllOptionalTypes()` would return the type itself if its not optional where as `getOptionalObjectType()` would return null - which I use as a check to ensure the type has at least one level of optionality. I am not sure if there's a better way to do it though?

I have updated the comment!

Reply:
I just mean that if you do `if (auto objectType = type->getOptionalObjectType()) {`, you can later do `objectType->lookThroughAllOptionalTypes()` and that function will be starting at a one-less-optional level.  It's not much, but it's slightly better.

Reply:
The comment change looks fine, thanks!

Reply:
@rjmccall Ah ok, that makes sense! Made the change now.

Reply:
Thanks, looks good!